Analytics/models/pin_location_data.py

```diff
@@ -0,0 +1,344 @@
+        :param sat_cnt: GPS Satellite count in use at time of measurement
+        :param fix_quality: GPS Fix quality ( 0: not fix, 1: GPS Fix, 2: DGPS, etc...)
+        :param signal_quality: RSSI of tracker signal
+        :param battery: Battery level in percentage full
+        :param value: Value to be stored for location
+        """
+
+        self.tracker_id = tracker_id
+        if isinstance(timestamp, str):
+            self.timestamp = datetime.strptime(timestamp, "%d/%m/%Y %H:%M")
+        else:
+            self.timestamp = timestamp
+        self.longitude = longitude
+        self.latitude = latitude
+        self.speed = speed
+        self.heading = heading
+        self.elevation = elevation
+        self.sat_cnt = sat_cnt
+        self.fix_quality = fix_quality
+        self.signal_quality = signal_quality
+        self.battery = battery
+        self.charger = charger
+        self.value = value
+
+    @staticmethod
+    def builder(tracker_id: str = "", timestamp: str = datetime.now().strftime("%d/%m/%Y %H:%M"),
+                longitude: float = 0.0, latitude: float = 0.0, speed: float = 0.0, heading: float = 0.0, elevation=0.0,
+                sat_cnt: int = 0, fix_quality: int = 0, signal_quality: int = 0,
+                battery: float = 0.0, charger: bool = False,
+                value: dict = {"o3": str(random.uniform(0, 100)), "no2": str(random.uniform(0, 100))}) -> db.Model:
+        """
+        Create New LocationData Instance. Fill in default values for missing arguments
+        :param tracker_id: Tracker Id number
+        :param timestamp: When the measurement was taken
+        :param longitude: GPS Longitudinal Coordinate in decimal degrees (DD.ddddddd)
+        :param latitude: GPS Latitudinal Coordinate in decimal degrees (DD.ddddddd)
+        :param speed: Velocity of reciever
+        :param heading: Heading (Direction of travel in degrees clockwise from North(0 degrees)
+        :param elevation: Height in meters above MSL (Mean Sea Level)
+        :param charger: Charging True or False
+        :param sat_cnt: GPS Satellite count in use at time of measurement
+        :param fix_quality: GPS Fix quality ( 0: not fix, 1: GPS Fix, 2: DGPS, etc...)
+        :param signal_quality: RSSI of tracker signal
+        :param battery: Battery level in percentage full
+        :param value: Value to be stored for location
+        :return: A new Instance of LocationData
+        """
+        return LocationData(tracker_id, timestamp, longitude, latitude, speed, heading, elevation, charger, sat_cnt,
+                            fix_quality, signal_quality, battery, value)
+
+    @property
+    def json(self) -> dict:
+        """
+        Get JSON encoded Attributes
+        :return:  JSON representation of LocationData instance
+        """
+        return {"id": self.id,
+                "tracker_id": self.tracker_id,
+                "timestamp": self.timestamp.strftime("%d/%m/%y %H:%M"),
```

Comment:
This should be `%d/%m/%Y %H:%M`

Reply:
All datetime format string standardize where possible.  
    -` '%d/%m/%Y %H:%M'` when the time is present in the string
    - `'%d/%m/%Y'` when the time is not present in the string